internal/service/device.go

```diff
@@ -148,23 +148,7 @@ func (h *ServiceHandler) ListDevices(ctx context.Context, request server.ListDev
 
 // (DELETE /api/v1/devices)
 func (h *ServiceHandler) DeleteDevices(ctx context.Context, request server.DeleteDevicesRequestObject) (server.DeleteDevicesResponseObject, error) {
-	allowed, err := auth.GetAuthZ().CheckPermission(ctx, "devices", "deletecollection")
-	if err != nil {
-		h.log.WithError(err).Error("failed to check authorization permission")
-		return server.DeleteDevices503JSONResponse{Message: AuthorizationServerUnavailable}, nil
-	}
-	if !allowed {
-		return server.DeleteDevices403JSONResponse{Message: Forbidden}, nil
-	}
-	orgId := store.NullOrgId
-
-	err = h.store.Device().DeleteAll(ctx, orgId, h.callbackManager.AllDevicesDeletedCallback)
-	switch err {
-	case nil:
-		return server.DeleteDevices200JSONResponse{}, nil
-	default:
-		return nil, err
-	}
+	return nil, fmt.Errorf("delete all operations on devices are not permitted")
```

Comment:
This seems to also throw an error 500 in the CLI and the error message is not displayed

![delete-all-devices](https://github.com/user-attachments/assets/1492002a-9c8d-4050-b883-2018962311e1)

Reply:
Is there a reason to keep this endpoint around if we're not implementing it?

Reply:
I would be fine removing it.

Reply:
Actually it looks like removing `DELETE` for devices would violate one of our `openapi` rules: https://github.com/flightctl/flightctl/blob/main/.spectral.yaml#L37. Thoughts?